### PR TITLE
[DOCS] Expand operator privileges explanation for users and roles

### LINKED ISSUE #82462: Clarify Operator Privileges implications for roles
1. Add a new paragraph to https://www.elastic.co/guide/en/elasticsearch/reference/current/operator-privileges.html that is a little more explicit about ESS/ECE that says something like  
   > When running Elasticsearch on Elastic Cloud (or Elastic Cloud Enterprise) Operator Privileges are enabled, which means that some infrastructure management functionality is restricted and cannot be accessed by your administrative users. This protects your cluster from unintended infrastructure changes.
2. Add a small note at the top of the builtin users page that refers to the main operator privileges docs  
   > When running on Elastic Cloud, ["operator privileges"](https://www.elastic.co/guide/en/elasticsearch/reference/current/operator-privileges.html) are enabled, and some infrastructure management functionality is restricted even if a role would otherwise permit the user to perform that task.
3. Add a very small comment in `superuser` like 
   > Note: On Elastic Cloud, all standard users, including those with `superuser` are restricted from performing [operator-only](https://www.elastic.co/guide/en/elasticsearch/reference/current/operator-privileges.html) actions

Maybe that's too much, but I think it's manageable, and it explicit about the behaviour where it is most likely to be read.

_Originally posted by @tvernum in https://github.com/elastic/elasticsearch/pull/82110#discussion_r781727835_

### PR DESCRIPTION
**Pull request by lockewritesdocs**
Operator privileges are enabled on Elastic Cloud, but our documentation currently doesn't describe what impact that behavior has on users and roles. This PR implements the suggestions from #82462 to clarify operator privileges, built-in roles, and the `superuser` role. The changes also provide more detail about the `elastic` user to align with the changes to the `superuser` role in #82110.

Preview links:

* [Operator privileges](https://elasticsearch_82893.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/operator-privileges.html)
* [Built-in users](https://elasticsearch_82893.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/built-in-users.html)
* [Built-in roles](https://elasticsearch_82893.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/built-in-roles.html)

Closes #82462

### CODE REVIEW

**Comment by lockewritesdocs** [comment]
@bytebilly, should these change also be applied to 7.16-7.17, with the exception that the descriptions for the `superuser` role and `elastic` user would need to change "ready-only access" to "write access"?

**Review by kirakoki — APPROVED** [review]

File: x-pack/docs/en/security/authentication/built-in-users.asciidoc [hunk 1]
```diff
@@ -7,7 +7,27 @@ up and running. These users have a fixed set of privileges and cannot be
 authenticated until their passwords have been set. The `elastic` user can be
 used to <<set-built-in-user-passwords,set all of the built-in user passwords>>.
 
+.Create users with minimum privileges
+****
+The built-in users serve specific purposes and are not intended for general
+use. In particular, do not use the `elastic` superuser unless full access to
+the cluster is absolutely required. Instead, create users that have the minimum
```
Comment (bytebilly) [inline x-pack/docs/en/security/authentication/built-in-users.asciidoc:14]:
```suggestion
the cluster is absolutely required. Instead, use it to create users that have the minimum 
```

Comment (bytebilly) [reply x-pack/docs/en/security/authentication/built-in-users.asciidoc:14]: Should we also specify "on self-managed deployments"?

**Review by bytebilly — CHANGES_REQUESTED** [review]
Thanks @lockewritesdocs, Ieft a comment. Everything else LGTM.

File: x-pack/docs/en/security/authentication/built-in-users.asciidoc (hunk 1 above)
Comment (lockewritesdocs) [reply x-pack/docs/en/security/authentication/built-in-users.asciidoc:14]: @bytebilly, we could change this to read:

>On self-managed deployments, use the `elastic` user to create users that have the minimum necessary roles or privileges for their activities.

Is the `elastic` user not available (or not created) on Elastic Cloud deployments?

Comment (bytebilly) [reply x-pack/docs/en/security/authentication/built-in-users.asciidoc:14]: suggestion works for me. On Cloud we still have `elastic`, but given there is single sign-on it's not the recommended way to authenticate for interactive users.

**Pull request merged**